### PR TITLE
Add Emoji Category Picker

### PR DESCRIPTION
**Pull request by stitesExpensify**
### Details
- Adds an emoji category picker to all platforms
- Updates Category Labels to match rest of app
- Updates Web/Desktop search bar to match rest of app
- Simplifies keyboard navigation offset calculations for web/desktop

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/239100

### Tests
*All Platforms*
1. Create a new account
2. Choose any chat
3. Open the emoji picker
4. Make sure there is a category picker row with only 8 categories
5. Click around on the different categories and make sure you are brought to the right place
6. Choose any emoji
7. Re-open the emoji picker and make sure that there are now 9 categories (including the frequently used category)
8. Click around on the different categories and make sure you are brought to the right place

-----------------------------
*Web / Desktop*
1. Choose any chat
2. Open the emoji picker
3. Search and make sure the results are what you would expect
4. Use the keyboard to navigate around the emojis and choose one
9. Make sure nothing looks out of the ordinary

- [x] Verify that no errors appear in the JS console

### Offline tests
n/a functionality is not affected by online/offline

### QA Steps
*All Platforms*
1. Create a new account
2. Choose any chat
3. Open the emoji picker
4. Make sure there is a category picker row with only 8 categories
5. Click around on the different categories and make sure you are brought to the right place
6. Choose any emoji
7. Re-open the emoji picker and make sure that there are now 9 categories (including the frequently used category)
8. Click around on the different categories and make sure you are brought to the right place

-----------------------------
*Web / Desktop*
1. Choose any chat
2. Open the emoji picker
3. Search and make sure the results are what you would expect
4. Use the keyboard to navigate around the emojis and choose one
5. Make sure nothing looks out of the ordinary

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/42391420/216422108-f3204a96-45c5-46de-b88d-84a66a927837.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/42391420/216422089-25999b89-991e-44ec-abae-109842e749d6.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/42391420/216422068-20bd9c82-f706-4342-8002-75d84e806085.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/42391420/216422346-f8732f36-81c8-4b20-a834-503a6eac949f.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/42391420/216422030-e87e0a75-eef8-4031-badf-870b53bb9f30.mp4

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/42391420/218188591-065b913a-4ab9-4994-8ea7-6bdca1f5ab75.mp4

</details>

### CODE REVIEW

**Comment by stitesExpensify** [comment]
@Pujan92 I saw that you fixed the issue of the emoji picker jumping in [this PR](https://github.com/Expensify/App/pull/13787) do you know what is going on here? The issue only appears to happen on mobile emulators.

iOS:
https://user-images.githubusercontent.com/42391420/214448522-9581084c-40f8-4af4-a89d-179cef955e96.mp4

**Comment by Pujan92** [comment]
> iOS: https://user-images.githubusercontent.com/42391420/214448522-9581084c-40f8-4af4-a89d-179cef955e96.mp4

@stitesExpensify Here as we are trying to scroll to the category header, we need to provide `getItemLayout` calculation to flatlist which skips the measurement of dynamic content. For the web I have added in that PR, you need to add the same for the native file.

```diff
diff --git a/src/components/EmojiPicker/EmojiPickerMenu/index.native.js b/src/components/EmojiPicker/EmojiPickerMenu/index.native.js
index bf9ac71bcf..cc4e46312a 100644
--- a/src/components/EmojiPicker/EmojiPickerMenu/index.native.js
+++ b/src/components/EmojiPicker/EmojiPickerMenu/index.native.js
@@ -111,6 +111,10 @@ class EmojiPickerMenu extends Component {
         this.emojiList.scrollToOffset({offset: testoffset, animated: false});
     }
 
+    getItemLayout(data, index) {
+        return {length: CONST.EMOJI_PICKER_ITEM_HEIGHT, offset: CONST.EMOJI_PICKER_ITEM_HEIGHT * index, index};
+    }
+
     /**
      * Given an emoji item object, render a component based on its type.
      * Items with the code "SPACER" return nothing and are used to fill rows up to 8
@@ -167,6 +171,7 @@ class EmojiPickerMenu extends Component {
                         this.isMobileLandscape() && styles.emojiPickerListLandscape,
                     ]}
                     stickyHeaderIndices={this.unfilteredHeaderIndices}
+                    getItemLayout={this.getItemLayout}
                 />
```

**Comment by stitesExpensify** [comment]
Ah, I see, thank you. For some reason I thought it was added to both. Going back and looking, I see that we said it was only happening on web/desktop which is interesting... I'm glad the solution works across the board though

**Comment by Pujan92** [comment]
I don't think it is due to emulator, FlatList renders partial emojis at a time for a performance, so I think animation here will be beneficial.

**Comment by stitesExpensify** [comment]
> I don't think it is due to emulator, FlatList renders partial emojis at a time for a performance, so I think animation here will be beneficial.

Why would it only happen on the emulators and not on web/desktop then?

**Comment by stitesExpensify** [comment]
Hmm it does seem to be happening on my physical device as well :/

**Comment by stitesExpensify** [comment]
It looks like scrollToIndex instead of scrollToOffset works _slightly_ better, but still not acceptable IMO

https://user-images.githubusercontent.com/42391420/215127811-e42335dc-9716-4588-831f-7d4ab0a9194a.mp4

**Comment by Pujan92** [comment]
> It looks like scrollToIndex instead of scrollToOffset works _slightly_ better, but still not acceptable IMO

I tried on the emulator but not seeing any difference, but yeah `scrollToIndex` can help to remove an offset calculation part.

**Comment by stitesExpensify** [comment]
Ready for review!

**Comment by PauloGasparSv** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by PauloGasparSv** [comment]
Will have to review this tomorrow.

@stitesExpensify I think you missed an unchecked item in the PR Author Checklist

**Comment by stitesExpensify** [comment]
For testing iOS if you have an iPhone, I'd recommend using [this QR code](https://github.com/Expensify/App/pull/14532#issuecomment-1412384986) because the emulator is really laggy, and the actual release app is not

**Comment by PauloGasparSv** [comment]
Hey @stitesExpensify, I see this strange behavior in Desktop where the flag icon is missing and the "Smiles & Emotions" category icon takes me to the "Animals & Nature" section.

I think it has to do with me having no "Frequently Used" category on desktop as my Web tests worked fine but I have that section on Web. Can you please take a look?

Desktop:
<video src="https://user-images.githubusercontent.com/6564265/216398920-a3f189b1-baff-425b-8eca-76f3bbee42c6.mov"/>

Web:
<video src="https://user-images.githubusercontent.com/6564265/216398936-2d82ea3d-b974-45a2-a173-ae2a77b6c727.mov"/>

**Comment by stitesExpensify** [comment]
Nice catch! I know exactly why that's happening. I'll get it fixed shortly

**Comment by stitesExpensify** [comment]
Updated! I updated the tests/QA sections, and all testing videos as well

https://user-images.githubusercontent.com/42391420/216418043-08316d69-838c-42d3-97c6-c14f271731e3.mp4

**Comment by PauloGasparSv** [comment]
Re-testing here!

**Comment by PauloGasparSv** [comment]
Hey @stitesExpensify does this happen to you on mobile web iOS too? If I click on a category button while scrolling things are not rendered correctly sometimes. I don't think this is a blocker as scrolling again fixes it and doesn't look like a problem in your logic.

https://user-images.githubusercontent.com/6564265/216445926-91184278-05cf-4982-81f2-df4d2427c6a8.mov

**Comment by stitesExpensify** [comment]
Yes I'm seeing that too. I tried to do something clever like disabling and then re-enabling scroll but that doesn't seem to help. Unfortunately there's no way to programmatically stop the momentum scroll, so I'm not sure what even a potential solution would be. Very weird.

**Review by PauloGasparSv — DISMISSED** [review]
LGTM! All yours @aimane-chnaif 

As a newDot user I loved these changes : )

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

**Comment by stitesExpensify** [comment]
Woah really? That's weird. Definitely LMK if you can reproduce consistently!

**Comment by stitesExpensify** [comment]
Updated to use flex, that's a way better solution. Good shout!

**Review by PauloGasparSv — DISMISSED** [review]
Everything still looking and working the same, nice!

**Comment by stitesExpensify** [comment]
Sounds good! Thanks!

**Comment by stitesExpensify** [comment]
Hi @aimane-chnaif! Any luck finding a consistent reproduction this weekend?

**Comment by stitesExpensify** [comment]
Okay we decided [here](https://github.com/Expensify/Expensify/issues/239100#issuecomment-1423145868) that we're going to go with reanimated. I'll push a PR tomorrow and we can get this merged after we re-test!

**Comment by stitesExpensify** [comment]
> reanimated library is updated now.

What does that mean @aimane-chnaif ?


Also, I updated native to reanimated, but just switched the web version to `animated=true` to match, and it fixed the iOS mobile web issue!

**Comment by stitesExpensify** [comment]
Oh, are you saying because it was updated, we need to test again with the new version?

**Comment by stitesExpensify** [comment]
Updated, and it still seems to be performing correctly!

**Comment by stitesExpensify** [comment]
This is now ready for re-review!

**Review by PauloGasparSv — APPROVED** [review]
LGTM!

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/6564265/218470321-281f2c4f-32ce-4ce7-888e-28323355fd2f.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/6564265/218477959-f09f498d-cfc6-4533-ad74-5b5bf521797b.mov


</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/6564265/218475473-f31ebe0f-87df-4c45-b6b3-d4bc8026c944.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/6564265/218470527-f0304bd5-dcea-48f1-a8c9-f7280a8e7c8f.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/6564265/218474966-31cc67b9-c33e-41c0-af32-e9721958c5a9.mov


</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/6564265/218477639-5168715c-2cfe-4812-a6d0-d5c323a74d5d.mov


</details>

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

File: src/components/EmojiPicker/EmojiPickerMenu/index.native.js [hunk 1]
```diff
@@ -101,7 +102,11 @@ class EmojiPickerMenu extends Component {
     scrollToHeader(headerIndex) {
         const calculatedOffset = Math.floor(headerIndex / CONST.EMOJI_NUM_PER_ROW) * CONST.EMOJI_PICKER_HEADER_HEIGHT;
         this.emojiList.flashScrollIndicators();
-        this.emojiList.scrollToOffset({offset: calculatedOffset, animated: false});
+        const node = findNodeHandle(this.emojiList);
+        runOnUI(() => {
+            'worklet';
```
[the remark this answers is not in the document]
Comment (PauloGasparSv) [reply src/components/EmojiPicker/EmojiPickerMenu/index.native.js:107]: Hey @aimane-chnaif IMO I don't think the comment will add value there. Will merge the P.R. here : )

**Pull request merged**